src/commands/Tools/Websearch/country.ts

```diff
@@ -0,0 +1,103 @@
+import { SkyraCommand, SkyraCommandOptions } from '@lib/structures/SkyraCommand';
+import { ApplyOptions } from '@skyra/decorators';
+import { fetch } from '@utils/util';
+import { KlasaMessage } from 'klasa';
+import { MessageEmbed } from 'discord.js';
+import { BrandingColors } from '@utils/constants';
+import { UserRichDisplay } from '@lib/structures/UserRichDisplay';
+import { DbSet } from '@lib/structures/DbSet';
+
+const mapNativeName = (data: { name: string; nativeName: string }) => `${data.name} ${data.nativeName === data.name ? '' : `(${data.nativeName})`}`;
+const mapCurrency = (currency: CurrencyData) => `${currency.name} (${currency.symbol})`;
+
+@ApplyOptions<SkyraCommandOptions>({
+	description: language => language.tget('COMMAND_COUNTRY_DESCRIPTION'),
+	extendedHelp: language => language.tget('COMMAND_COUNTRY_EXTENDED'),
+	usage: '<country:str>'
+})
+export default class extends SkyraCommand {
+
+	public async run(message: KlasaMessage, [countryName]: [string]) {
+		const response = await message.sendEmbed(new MessageEmbed()
+			.setDescription(message.language.tget('SYSTEM_LOADING'))
+			.setColor(BrandingColors.Secondary));
+
+		const countries = await this.fetchAPI(message, countryName);
+		if (countries.length === 0) throw message.language.tget('SYSTEM_QUERY_FAIL');
+
+		const display = await this.buildDisplay(message, countries);
+		await display.start(response, message.author.id);
+		return response;
+	}
+
+	private async fetchAPI(message: KlasaMessage, countryName: string) {
+		const apiResult = await fetch<CountryApiResult>(`https://restcountries.eu/rest/v2/name/${encodeURIComponent(countryName)}`)
+			.catch(() => { throw message.language.tget('SYSTEM_QUERY_FAIL'); });
+		if (Reflect.has(apiResult, 'status') && (apiResult as CountryResultError).status === 404) {
+			throw message.language.tget('SYSTEM_QUERY_FAIL');
+		}
```

Comment:
Does the API return 200 OK on a 404? `utils.fetch` throws on 400-599 status codes.

Reply:
It returns 404 NOT FOUND, I'll remove the check.